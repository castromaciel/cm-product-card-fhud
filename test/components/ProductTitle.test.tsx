import React from 'react';
import renderer from 'react-test-renderer'
import { ProductCard } from '../../src/components/ProductCard';
import { ProductTitle } from '../../src/components/ProductTitle';
import { product1 } from '../data/data';


describe('ProductTitle', () => {
  test('Debe de mostrar el componente correctamente con el titulo personalizado', () => {

    const wrapper = renderer.create(
      <ProductTitle title='custom product' className='custom-class'/>
    )

   expect( wrapper.toJSON() ).toMatchSnapshot();
  })

  test('Debe de mostrar el componente con el nombre del producto',() => {
    const wrapper = renderer.create(
      <ProductCard product={product1}>
        {
          () => (
            <ProductTitle />
          )
        }
      </ProductCard>

    )

   expect( wrapper.toJSON() ).toMatchSnapshot();

  })
})