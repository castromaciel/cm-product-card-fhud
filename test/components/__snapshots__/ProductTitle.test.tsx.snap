// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle Debe de mostrar el componente con el nombre del producto 1`] = `
<article
  className=" productCard undefined"
>
  <span
    className="productDescription undefined"
  >
    Coffe Mug - Card
  </span>
</article>
`;

exports[`ProductTitle Debe de mostrar el componente correctamente con el titulo personalizado 1`] = `
<span
  className="productDescription custom-class"
>
  custom product
</span>
`;
